firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -436,8 +434,8 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
         store.dispatch(action)
     }
 
-    @MainActor
     private func selectTabWithSession(tab: Tab, previous: Tab?, sessionData: Data?) {
+        assert(Thread.isMainThread, "Currently expected to be called only on main thread.")
```

Comment:
Nice! Can we add another assert to `BrowserViewController.swift` line 3470 to check that the `selected` Tab has a non-nil `webView` as well? That caused all the other problems we saw.